crates/sui/src/client_commands.rs

```diff
@@ -425,6 +426,19 @@ impl SuiClientCommands {
                     },
                 )?;
 
+                if !compiled_package.is_framework() {
```

Comment:
I kept this check in because it exists in the original check, but I think it might be unnecessary?  Presumably if the package is the framework package, publishing it is going to fail for some other reason?

Reply:
Indeed, I think publishing the framework should fail do the "`0x0` self ID check". But we should verify.

Reply:
In case I've misunderstood -- isn't this the "`0x0` self ID check", or do you mean the one on the validator?

I think the special case for framework modules was previously necessary because we would use the same code path to compile it as all other modules, but it was published as part of genesis and not the sui client publish command, so now, we should be able to remove this check, because we never try to publish the framework this way with the intention of it succeeding, so it's fine that it would fail on the client, rather than go to the validator and fail there.